### ISSUE TITLE
potential bug in sample_pseudoabs

### ISSUE THREAD

**Issue by pitherj** [EvolEcolGroup/tidysdm#24]
Thanks for this excellent contribution!

This is my first attempt at submitting an "issue", so apologies if i'm not doing this right.  

I'm using R version 4.3.1 on MacOS Ventura 13.6.1

Following the tidysdm vignette (https://cran.r-project.org/web/packages/tidysdm/vignettes/a0_tidysdm_overview.html) and all's fine until the first use of the `sample_pseudoabs` function. 

Looking into the function itself, it seems to be having issues when the function is provided with a standard "sf" object (one that has geometry, but no separate columns for coordinates).  The function detects this and alters the input sf object by binding new "X" and "Y" columns, but further down in the function this causes an issue when attempting to "bind_rows" with the newly generated pseudoabsences.

here's the error:

Error in `dplyr::bind_rows()`:
! Can't combine `..1$class` <character> and `..2$class` <double>.
---
Backtrace:
    ▆
 1. ├─tidysdm::sample_pseudoabs(...)
 2. │ └─presences %>% dplyr::bind_rows(pseudoabsences) %>% ... at <tmp>:73:8
 3. ├─dplyr::mutate(., class = stats::relevel(factor(class), ref = "presence"))
 4. └─dplyr::bind_rows(., pseudoabsences)
Run rlang::last_trace(drop = FALSE) to see 10 hidden frames.

**Comment by dramanica** [comment]
Hi Jason, a couple of questions, so that we can pin down what is causing the error. Are you using the built in dataset lacerta, like in the vignettes? Or are you using your own data? Either way, could you post a `summary()` output of the `sf` object you are feeding the function?

**Comment by pitherj** [comment]
I'm truly sorry - I've discovered my rather novice mistake. I incorrectly specified the "method" argument.   You can remove this issue!  Sorry for the bother!

**Comment by dramanica** [comment]
Not a problem, but a well behaved function should have been a bit more helpful with its error message. If you don't mind sharing the incorrect line of code, I can make sure that we do a better job at catching a mispecified method.

**Comment by pitherj** [comment]
Certainly! 
The offending code was the ")" after "dist_min": 
```
sample_pseudoabs(lacerta_thin, 
                                 n = 3 * nrow(lacerta_thin), 
                                 raster = land_mask, 
                                 method = c("dist_min"), km2m(50))
```

The correct code:
```
sample_pseudoabs(lacerta_thin, 
                                 n = 3 * nrow(lacerta_thin), 
                                 raster = land_mask, 
                                 method = c("dist_min", km2m(50)))
```

**Comment by dramanica** [comment]
Thanks @pitherj  for helping fixing this. In the `dev` version of `tidysdm`, the incorrect code would now give the error:

`method 'dist_min' should have one threshold, e.g, c('dist_min',10)`

which hopefully should give a hint of what the problem is.

Fixed by b51acbfcc193fe232df6da1502e04f998f641fdd

**Comment by pitherj** [comment]
thank you!
